### PR TITLE
Consolidated README-docs and clean-up of documentation

### PR DESCRIPTION
**Pull request by CGMossa**
Our documentation needs to improve.
I'm also reading up on what is possible with Extendr at the same time and cleaning up some things here and there.

`cargo doc` passes now without any warnings.

Instead of duplicating the `README.md` in workspace crates, and in `lib.rs`  therein,
I've used a new feature of `rustdoc` to embed the `README.md` in the `lib.rs`.

Here I had to make a choice, and I favoured making clickable references in the documentation.

I hope ya'll agree with this.

### CODE REVIEW

File: extendr-engine/README.md
Comment (multimeric) [inline extendr-engine/README.md:12]: This line is a bit redundant.

**Comment by multimeric** [comment]
This looks great! The sub-crate documentation is very helpful, as are the clickable links, and fixing typos.

**Review by multimeric — APPROVED** [review]

File: extendr-engine/README.md
Comment (CGMossa) [reply extendr-engine/README.md:12]: I think we should change this description in another PR.
I didn't write this myself :(
I just need a little more time with the codebase.. 
Is it okay I mark this resolved?

Comment (multimeric) [reply extendr-engine/README.md:12]: Oh okay.

**Comment by CGMossa** [comment]
I don't dare to merge when CI is failing. @Ilia-Kosenkov or @yutannihilation are you confident that I can merge despite this?
Feel free to merge this for me if you're feeling up for it.

**Comment by Ilia-Kosenkov** [comment]
Half of the errors are due to runner's inability to download R, which is strange but should resolve itself.
However, this is a real issue
```
   thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseError(["~"])', src/lib.rs:5:1
```

**Comment by CGMossa** [comment]
Status here, I suppose, is that there are code that was written in `README`s that didn't actually compile.
That's not good, and it is hard point it out, but I guess that issue needs to be dealt with, before this can be merged.

**Pull request closed without merging**
